src/vmm/src/guest_config/cpuid/amd/normalize.rs

```diff
@@ -408,3 +426,52 @@ impl super::AmdCpuid {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use std::collections::BTreeMap;
+
+    use super::*;
+    use crate::guest_config::cpuid::AmdCpuid;
+
+    #[test]
+    fn test_update_structured_extended_entry() {
+        // Case 1: Leaf 0x7 / subleaf 0x0 does not exist.
+        let mut cpuid = AmdCpuid(BTreeMap::new());
```

Comment:
nitpick: Would it be possible to use a prebuilt CPUID instance from `static_cpu_templates::t2a`?

Reply:
`static_cpu_templates::t2a` uses no longer the previous exhaustive style, because they can be backward incompatible with static templates we have in the main branch (using the style where it only modifies needed bits/subleaves/leaves). Thus, it is impossible to use static CPU templates to construct `AmdCpuid`.

Reply:
I understand. The static template would still pre-initialize the structure for you, and you could modify the tree to add the leaf for case 2 when you need it though. Not a big deal though.

Reply:
You're right. If I use `CpuConfiguration::new()` with static template, the CPUID info is populated. I don't have a concrete good implementation at the moment. Could you please provide me the actual code sample if you have? I am willing to accept the sample!